zerver/lib/export.py

```diff
@@ -434,8 +433,7 @@ def floatify_datetime_fields(data: TableData, table: TableName) -> None:
                 continue
             assert isinstance(dt, datetime.datetime)
             assert not timezone_is_naive(dt)
```

Comment:
Does this assert still make sense here? 

Also, do our tests verify this produces the same output as before? The previous logic is sufficiently convoluted that I'm not entirely sure.

Reply:
> Does this assert still make sense here?

For a naive `datetime` object, `.timestamp()` implicitly assumes it’s local and adjusts it for the UTC offset. That might or might not be expected, so we might as well continue asserting against it.

We could replace the check with `dt.utcoffset() is not None` if we want to avoid the unnecessary Django wrapper.

> Also, do our tests verify this produces the same output as before? The previous logic is sufficiently convoluted that I'm not entirely sure.

Yes, a couple of tests fail if this computation is disturbed: `zerver.tests.test_import_export.RealmImportExportTest.test_import_realm` and `zerver.tests.test_import_export.SingleUserExportTest.test_user_data`.

I suspect the previous logic had some incorrect edge cases during DST transitions though.